lib/improver/tests/nowcasting/nowcasting/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,457 @@
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.25
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_half_past(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at half past an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.51
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.second_slice.metadata)
+        self.assertEqual(result, self.second_slice)
+        self.assertEqual(result.coord("time"), self.second_slice.coord("time"))
+
+    def test_alternative_time_quarter_to(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at quarter to an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.75
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.second_slice.metadata)
+        self.assertEqual(result, self.second_slice)
+        self.assertEqual(result.coord("time"), self.second_slice.coord("time"))
+
+
+class Test__apply_cube_combiner(IrisTest):
+
+    """Test the __apply_cube_combiner method."""
+
+    def setUp(self):
+        """Set up cubes for testing."""
+        self.precip_cubes = set_up_precipitation_rate_cube()
+        self.oe_cubes = set_up_orographic_enhancement_cube()
+
+    def test_check_expected_values_first_add(self):
+        """Test the expected values are returned when cubes are added.
+        First check."""
+        expected0 = np.array([[[[0., 1., 2.],
+                                [1., 2., 3.],
+                                [0., 3., 4.]]]])
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin._apply_cube_combiner(
+            self.precip_cubes[0], self.oe_cubes[0])
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.precip_cubes[0].metadata)
+        result.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result.data, expected0)
+
+    def test_check_expected_values_second(self):
+        """Test the expected values are returned when cubes are combined.
+        Second check."""
+        expected1 = np.array([[[[6., 5., 1.],
```

Comment:
Can we lose the 1 please (as above, and also for the tests that follow).

Reply:
What is the difference between this test and the one above?  They look identical to me (other than the input data arrays).  If so we don't need this one (as the other test for correct treatment of zeroes).

Reply:
Removed this particular test.